src/Athena/Controllers/api/OfferingController.cs

```diff
@@ -0,0 +1,114 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Athena.Core.Repositories;
+using Athena.Core.Models;
+using Athena.Exceptions;
+using System.Net;
+
+namespace Athena.Controllers.api
+{
+    [Route("api/v1/[Controller]")]
+    public class OfferingController : Controller
+    {
+        private readonly IOfferingReository offerings;
+        private readonly IMeetingRepository meetings;
+        private readonly ICourseRepository courses;
+
+        public OfferingController(IOfferingReository offeringsRepository, IMeetingRepository meetingsRepository, ICourseRepository coursesRepository)
+        {
+            offerings = offeringsRepository ?? throw new ArgumentNullException(nameof(offeringsRepository));
+            meetings = meetingsRepository ?? throw new ArgumentNullException(nameof(meetingsRepository));
+            courses = coursesRepository ?? throw new ArgumentNullException(nameof(coursesRepository));
+        }
+        [HttpPost]
+        public async Task AddOffering([FromBody] Offering offering) => await offerings.AddAsync(offering);
+        
+        [HttpGet("{id}")]
+        public async Task<Offering> GetOffering(Guid id) => await offerings.GetAsync(id);
+
+        [HttpPut("{id}")]
+        public async Task EditOffering(Guid id, [FromBody] Offering offering)
+        {
+            if (id != offering.Id)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to edit {id} but got a model for {offering.Id}");
+            }
+            await offerings.EditAsync(offering);
+        }
+
+        [HttpDelete("{id}")]
+        public async Task DeleteOffering(Guid id)
+        {
+            var offering = await offerings.GetAsync(id);
+            if (offering == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to delete {offering} that does not exist");
+            }
+            await offerings.DeleteAsync(offering);
+        }
+
+        [HttpGet("api/v1/course/{id}/offering/{offeringId}")]
+        public async Task<IEnumerable<Offering>> GetOfferingsForCourseAsync(Course course)
+        {
+            if (course == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to get offering for {course} that doesn't exist");
+            }
+            return (await offerings.GetOfferingsForCourseAsync(course));
+        }
+
+        [HttpPost("api/v1/course/{id}/offering/{offeringId}/meeting")]
+        public async Task AddMeetingAsync(Offering offering, Meeting meeting)
+        {                                       
+            if (meeting == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to add meeting {meeting} where meeting does not exist");
+            }
+            await offerings.AddMeetingAsync(offering, meeting);
+        }
+
+        [HttpDelete("api/v1/course/{id}/offering/{offeringId}/meeting")]
```

Comment:
Did you intend to implement this here or in a composite controller? If you intend to implement it here you need to use `/api/v1/course...` to override the controller-wide prefix.